uaa/src/main/webapp/WEB-INF/spring/oauth-clients.xml

```diff
@@ -82,7 +82,7 @@
                             <entry key="secret" value="loginsecret"/>
                             <entry key="scope" value="openid,oauth.approvals"/>
                             <entry key="authorized-grant-types" value="client_credentials,authorization_code"/>
-                            <entry key="redirect-uri" value="http://localhost/**,http://localhost:7000/**"/>
```

Comment:
What is the purpose of this change? Is this required for the added test?

Reply:
it mainly is the revert of my change https://github.com/cloudfoundry/uaa/pull/2403/files#diff-d02078cfeded6b34f4eb0a68b222758f944db9bbe2fa1227249a9c782718cf09R85

the login client is used in our internal forked scenario and it was broken. Because UAA has this hidden feature of redirect uris. It means if you have only one redirect uri defined you can omit the parameter redirect_uri in authorize request.
If you have an array in the client you cannot use this feature anymore therefore more the 2nd entry

Reply:
and create an explicit test for this hidden feature